deployment/helm/templates/pre-delete-job.yaml

```diff
@@ -0,0 +1,36 @@
+{{- if .Values.cleanupHook.enabled }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ .Release.Name }}-cleanup"
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+  annotations:
+    # This is what defines this resource as a hook. Without this line, the
+    # job is considered part of the release.
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      name: "{{ .Release.Name }}"
+      labels:
+        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        app.kubernetes.io/component: helm-hook
+    spec:
+      restartPolicy: Never
+      serviceAccountName: akri-helm-cleanup-hook-sa
+      containers:
+      - name: delete-configurations
+        image: bitnami/kubectl:latest
+        command: ["/bin/sh", "-c"]
+        args: [
+          "kubectl delete configurations.akri.sh --all --all-namespaces"
```

Comment:
I think we should get it to wait for the deletion to really occur, otherwise we may end up marking it for deletion and helm then deleting the agents daemonset before the delete being processed.

Reply:
Good call. Just updated